runtime/service.go

```diff
@@ -143,6 +143,8 @@ type service struct {
 	driveMountStubs          []MountableStubDrive
 	exitAfterAllTasksDeleted bool // exit the VM and shim when all tasks are deleted
 
+	taskCleanupStack map[string][]func(context.Context) error
```

Comment:
Let's keep it simple. We right now don't need an arbitrary function and one task wouldn't need multiple cleanup functions.